lib/spack/spack/audit.py

```diff
@@ -416,12 +416,10 @@ def _check_build_test_callbacks(pkgs, error_cls):
         pkg_cls = spack.repo.PATH.get_pkg_class(pkg_name)
         test_callbacks = getattr(pkg_cls, "build_time_test_callbacks", None)
 
-        # TODO (post-34236): "test*"->"test_*" once remove deprecated methods
-        # TODO (post-34236): "test"->"test_" once remove deprecated methods
-        has_test_method = test_callbacks and any([m.startswith("test") for m in test_callbacks])
+        has_test_method = test_callbacks and any([m.startswith("test_") for m in test_callbacks])
```

Comment:
🤔 I know I flagged it for change (`test` ->  `test_`) in the mentioned PR but I do wonder if it is a good idea since I can imagine ppl being confused if their `test` method doesn't get executed. Definitely something to bring up in a discussion with others.

Reply:
Upon further reflection, perhaps it is fine to keep this change.